### PR TITLE
fix(collections): the header button syncs a googleCalendar collection (#2108)

### LINKED ISSUE #2108: googleCalendar コレクションの「同期」ボタンが "is not a feed (no ingest config)" で失敗する
## 起きたこと

`googleCalendar` ブロックを持つコレクション（`ingest` は持たない）を開き、ヘッダーの「同期」ボタンを押すと、次のエラーが出て同期できません。

```
読み込みに失敗しました: collection '<slug>' is not a feed (no ingest config)
```

毎時のシステムタスク（google-calendar-sync）による取り込みは動いており、`data/calendar/.sync-state.json` の `lastSyncedAt` は更新されています。できないのはボタンからの手動同期だけです。「Googleへ書き出し」（push）は別の経路で、この症状とは関係ありません。

これが不具合なのか、MulmoTerminal では手動同期を意図的に持たないのか、こちらでは判断できていません。

## 期待したこと

ボタンを押すと、そのコレクションが読むカレンダーの同期がその場で走り、レコードが再読み込みされる。もしくは、MulmoTerminal が手動同期に対応しないのであれば、ボタンが出ない。

## 再現手順

1. Google アカウントを連携する
2. `googleCalendar`（`calendarId` と `map`）を宣言し、`ingest` を持たないコレクションを用意する
3. MulmoTerminal でそのコレクションを開く
4. ヘッダーの「同期」ボタンを押す
5. 上記のエラーが表示される（毎回）

## 読んだ範囲で分かったこと

- `@mulmoclaude/collection-plugin` 1.1.0 から、`googleCalendar` を持つコレクションにも同じボタン（ラベル `collectionsView.syncCalendar`）が出ます。押すと `refreshCollection(slug)` 経由で `POST /api/collections/:slug/refresh` を呼びます（`CollectionHeader` の `refreshLabel`、`CollectionView` の `refreshFeed`）。表示の条件は schema だけで、ホスト側からボタンを隠す手段は見当たりませんでした。
- MulmoTerminal の `server/backends/feeds.ts` の refresh ルートは、`schema.ingest` が無いと 400 を返します。この分岐は #129 から変わっていません。
- MulmoClaude 側は receptron/mulmoclaude#2566（#2427）で `server/api/routes/collectionCalendarRefresh.ts` を足し、`@mulmoclaude/core/google` の `syncCalendarForCollection` で手動同期しています。MulmoTerminal のソースには `syncCalendarForCollection` の呼び出しがありません（origin/main でも同じ）。
- `docs/mulmoclaude-parity.md` では `refresh` は両ホストで同じパスとして挙げられています。
- MulmoTerminal が collection-plugin を 1.1.1 に上げたのは 78018921 です。

## 環境

- mulmoterminal 4.26.0（ローカル clone の main、246a5910）
- @mulmoclaude/collection-plugin 4.7.1 / @mulmoclaude/core 4.9.4
- Node v24.18.1
- macOS 26.6.2（Darwin arm64）
- 起動方法: ローカル clone から起動（`~/Developer/mulmoterminal`）

### PR DESCRIPTION
**Pull request by isamu**
## Summary

The shared collection plugin offers ONE header button with two labels — "Refresh" when a schema
declares `ingest`, "Sync" when it declares `googleCalendar` — and posts to
`POST /api/collections/:slug/refresh` either way. This host only ever answered the first, so every
press on a calendar collection came back `400 collection '<slug>' is not a feed (no ingest config)`,
and nothing lets a host hide the button.

mulmoclaude `be723e42b` (#2427) shipped both halves in one commit: the plugin's header, and
MulmoClaude's own refresh route arm. The host arm lives in each host's repository, so bumping the
plugin here (`78018921`) brought the button and nothing behind it. `git log --all -S
"syncCalendarForCollection"` is empty in this repo — never implemented, not regressed.

- `common/collectionRefresh.ts` — the wire shape BOTH arms answer, as `common/collectionPush.ts`
  already does for the push direction.
- `server/backends/calendarRefreshResult.ts` — pure; ports MulmoClaude's `calendarRefreshBody`,
  wording included, so someone running both hosts over one workspace gets one explanation per
  setup problem.
- `server/backends/calendarRefresh.ts` — the arm, engine injectable.
- `server/backends/feeds.ts` — the gate becomes "neither block"; `ingest` still wins when a schema
  declares both.
- `src/composables/collectionUi.ts` — the binding is typed against that shape, so the server's
  response and the view's expectation are checked against each other rather than described twice.
- `docs/mulmoclaude-parity.md` — the calendar wiring gets a row, and the remaining half of #2427
  gets an entry, so the next reader can see the gap this PR closes and the one it does not.

Design notes are in `plans/fix-2108-calendar-collection-refresh.md`.

## Items to Confirm / Review

1. **`calendarSyncEnabled` deliberately does not gate the button.** The setting is read only where
   the scheduled tasks are built, the manual feed refresh beside it does not consult
   `feedRefreshEnabled` either, and the Settings section calls itself "Built-in scheduled tasks …
   Both run hourly". So the toggle stops the hourly sync and the button still works. MulmoClaude
   has no equivalent setting, so this one was decided here rather than ported — please sanity-check
   the reading.
2. **The root is the request's, not the workspace's.** MulmoClaude syncs against one module-level
   `workspacePath`; this host serves several roots, so the arm is handed the same scope the
   `loadCollection` beside it used. A deliberate divergence, commented at the call.
3. **The 400 message changed text** (`is not a feed (no ingest config)` →
   `is not refreshable (no ingest or googleCalendar config)`). Nothing in this repo matched the old
   string, but it is user-visible.
4. **A live sync was not exercised.** No Google account is linked in the environment used, so the
   `kind: "synced"` path — real events, `removed`, the group fan-out — is covered by specs over the
   engine's declared outcome type, not by a real calendar. Everything else was driven against a
   running server (below). Worth one manual press by someone with a linked account.
5. **Codex cross-review ran to a clean round at tier C** (`/codex-cross-review`): three rounds,
   ledger and every exchange posted in this thread. Round 2 raised one correct P2 — the multi-root
   tests did not pin the contract they claimed — which is fixed and break-verified. Round 3 is LGTM
   on the head that will merge, with all ten axes answered. **CodeRabbit has reviewed no commit on
   this PR** (rate limited on every push), and the repo's Codex auto-review workflow is deliberately
   off, so Codex is the only reviewer that saw the final head.
6. **The second commit came out of my own review, not a bot's.** `common/collectionRefresh.ts` was
   sitting in `common/` on the grounds that both sides decide from it while nothing in `src/`
   actually read it. Typing the binding against it made the two meet — and they disagreed: the
   optionals said `| undefined`, which under `exactOptionalPropertyTypes` is not the plugin's
   `removed?: number`. The feed arm now omits keys it has no value for, which is what the wire
   carried anyway. Breaking a field in the shared type is 11 type errors across the server arms and
   the binding; before that commit the binding contributed none.
7. **Out of scope, but noticed:** `calendarPush.ts` justifies answering 200-on-refusal with "our
   `fetchJson` reports the bare `HTTP 400` and drops the body". `#913` (`e20ee14b`) taught
   `fetchJson` to read the body, an hour after that comment was written (`e268f5c1`). The route is
   not wrong; the reason given for it no longer holds. Not touched here.

## Verification

Reproduced and then re-driven against a real server on a throwaway workspace (scratch `HOME`,
nothing of the user's touched), one collection per case:

| request | before | after |
| --- | --- | --- |
| `googleCalendar`, no `ingest` | 400 `is not a feed (no ingest config)` | 200 `errors: ["no Google account is linked on this host — link it in Settings → Google"]` |
| declares neither | 400 `is not a feed …` | 400 naming both blocks |
| `ingest: rss` | 200 through the feeds engine | unchanged (the engine's own SSRF refusal came back, so the arm is intact) |
| unknown slug | 404 | unchanged |
| `calendar-push` | 200 + reason | unchanged |

The first row's "after" is the same sentence the push button gives for the same condition, which is
the parity constant holding at runtime rather than only in a spec. Which slot the view renders it in
follows from the plugin's own source: a non-2xx sets the page-level `loadError` (the
"読み込みに失敗しました: …" the reporter saw), a 200 with `errors` sets the banner beside the button.

`yarn typecheck` and `yarn lint` are clean over the whole repo and `npx vitest run
test/server/backends/` is green — run those to reproduce rather than trusting a count here. The
machine was heavily loaded, so the rest of the suite is left to CI, which is the ground truth those
commands approximate anyway.

The HTTP table above was taken against the first commit. Nothing since changes a response: the
second commit only decides whether an absent key is omitted or set to `undefined`, which
`JSON.stringify` already collapsed — proved by a 320-case differential of the old body expression
against the new one, identical wire bytes in every case — and the rest is tests and prose.

**Two claims are now pinned by tests rather than argued.** `dispatched: false` is a value the engine
sends, so the arm keys on `=== undefined` and not on truthiness; and both the lookup and the calendar
arm resolve the request's project root. Each was break-verified — reverting the sync to a module
workspace fails one test, reverting the lookup fails two, and rewriting the `dispatched` condition as
a truthiness check fails exactly the test written for it.

## User Prompt

- 調査して https://github.com/receptron/mulmoterminal/issues/2108
- （調査結果を受けて）issue にコメントして修正 PR まで

Closes #2108

work in mulmoterminal3

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by isamu** [comment]
## Review ledger — after iteration 1

Findings raised so far and what happened to each. Re-flagging a `FIXED` or `REJECTED` row is
welcome **if the resolution is wrong** — say which row and what is wrong with it.

| # | iter | source | finding (one line) | disposition | why / where |
|---|------|--------|--------------------|-------------|-------------|
| 1 | 1 | Claude (self-review, no bot flagged it) | `common/collectionRefresh.ts` claimed both sides decide from it while nothing in `src/` read it, so the response shape and the view's expectation were still described twice | FIXED | commit `cee35741` — the binding is typed against it; the two then disagreed on optional-vs-`undefined` and the wire shape was corrected to the plugin's |
| 2 | 1 | Claude (self-review) | Is `calendarSyncEnabled` supposed to gate the manual button? | REJECTED | The setting is read only where the scheduled tasks are built; the manual feed refresh beside it does not consult `feedRefreshEnabled` either, and the Settings section is titled "Built-in scheduled tasks … Both run hourly". Recorded in `plans/fix-2108-calendar-collection-refresh.md` |
| 3 | 1 | Claude (self-review) | `calendarPush.ts` justifies its 200-on-refusal divergence with a claim `fetchJson` has not matched since `#913` | DEFERRED | Prose, not a defect, and a different route than this PR's — raised as item 6 in the PR body rather than changed here |

**Reviewer status on this head:** CodeRabbit reports "Review rate limited" and has not read either
commit. The repo's Codex auto-review is deliberately off. So no bot has reviewed this PR at all —
that is an absence, not a sign-off.

**Comment by isamu** [comment]
CODEX VERDICT: LGTM

| Axis | Result |
| --- | --- |
| correctness and edge cases | none |
| security (path traversal / SSRF / injection / secrets / project-scope escape) | none |
| tests: does a test go red if the new behaviour breaks | none — targeted Vitest passed (25 tests); typecheck passed |
| API / schema / wire / on-disk compatibility | none |
| consistency with the rest of the codebase | none |
| accessibility and i18n lockstep | none |
| [C] BLAST RADIUS — schema shapes not considered | none — `ingest` retains precedence, `googleCalendar` is handled, and neither is rejected |
| [C] STATUS vs FIELD | none — the undeclared case remains HTTP 400 like the reference refresh route; sync refusals are HTTP 200 with `errors`, matching the view’s banner behavior |
| [C] MULTI-ROOT CORRECTNESS | none — the request-resolved root is used for both `loadCollection` and `syncCalendarCollection` |
| [C] CROSS-HOST PARITY | none — refresh path, status, wording, and wire fields match MulmoClaude; the omitted initial-calendar-sync half is explicitly documented as out of scope |

FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.

**Comment by isamu** [comment]
## Codex exchange — round 1 (tier C)

Tier **C**, set before the round and stated here for the record. Two C-list triggers: the route is
the entry point every collection refresh passes through, and the PR introduces a wire shape
(`common/collectionRefresh.ts`) and changes a user-visible 400 message.

**Sandbox pre-flight** (ledger row 0) — Codex can run vitest (exit 0), `yarn typecheck` (exit 0),
bind a socket (exit 0), reach api.github.com (200), and use `gh`. **File deletion is refused by
policy**, so the prompt explicitly forbids a mutation sweep — a failed restore would make every
later count measure the harness. Any mutation Codex names, I run.

<details><summary>Prompt sent (verbatim)</summary>

```text
Review PR #2111 at https://github.com/receptron/mulmoterminal/pull/2111.
Repo is checked out at /Users/isamu/ss/llm/mulmoterminal3 on branch fix/2108-calendar-collection-refresh.
The diff is `git diff 19a886acc47720fee85b6b5cf34c831bdaf3acca...HEAD`.

── THIS IS THE ONLY ROUND YOU ARE GUARANTEED ────────────────────────────────
Every finding you hold back costs a full extra round: a fix is reviewed for the
first time in the round AFTER it is pushed, so a finding raised late is not a
later finding, it is a repeat of this entire exchange. Read the WHOLE diff
before you post anything. Do not stop at the first problem. Do not save
anything for 'a follow-up review'.

Use the gh CLI to inspect the PR, then post inline review comments on specific lines via:
  gh api repos/receptron/mulmoterminal/pulls/2111/comments
or top-level comments via:
  gh pr comment 2111

── CONTEXT YOU WILL WANT ────────────────────────────────────────────────────
This is TIER C. Two reasons: the route changed is the entry point EVERY
collection refresh passes through, and the PR introduces a wire shape
(`common/collectionRefresh.ts`) plus changes a user-visible 400 message.

The bug: the shared @mulmoclaude/collection-plugin shows ONE header button with
two labels — "Refresh" for a schema declaring `ingest`, "Sync" for one declaring
`googleCalendar` — and posts to POST /api/collections/:slug/refresh either way.
This host only answered the `ingest` case, so a calendar collection got a 400.

MulmoClaude is the REFERENCE HOST and its source is a sibling checkout at
/Users/isamu/ss/llm/mulmoclaude. Its counterpart code is
  server/api/routes/collections.ts        (the refresh route, both arms)
  server/api/routes/collectionCalendarRefresh.ts  (the pure body shaping)
  server/services/google/initialCalendarSync.ts   (the half NOT ported here)
Compare against it directly — divergence from that host is a finding unless the
diff states a reason.

Do NOT run a mutation sweep: `rm` may be refused by the sandbox policy, and a
failed restore makes every later count measure the harness instead of the code.
If you want a mutation run, name it and I will run it.
Test invocation that works here: `npx vitest run <path>`. Typecheck: `yarn typecheck`.

── AXES — answer EVERY line, even when the answer is 'no findings' ──────────
  correctness and edge cases
  security (path traversal / SSRF / injection / secrets / project-scope escape)
  tests: does a test go red if the new behaviour breaks
  API / schema / wire / on-disk compatibility
  consistency with the rest of the codebase
  accessibility and i18n lockstep
  [C] BLAST RADIUS — every collection refresh passes through this route. What
      breaks for a schema shape the diff did not consider?
  [C] STATUS vs FIELD — which failures answer an HTTP status and which ride a
      200 with `errors`. The view routes the two to different places (page-level
      error slot vs the banner beside the button), so this is behaviour. Is the
      400 correct, and is it consistent with the sibling `calendar-push` route
      in server/backends/calendarPush.ts, which answers 200 for its refusals?
  [C] MULTI-ROOT CORRECTNESS — this host serves several project roots via
      `resolveProjectRoot(req)`; MulmoClaude serves one module-level workspace.
      Does every call in the new path use the REQUEST's root, and can a slug
      resolved in one root be synced against another?
  [C] CROSS-HOST PARITY — read ../mulmoclaude's counterpart files named above.
      Report every divergence in path, status, wording, or wire field, and say
      whether the diff justifies it.

An axis with no findings is a RESULT and has to be stated. 'No findings' on an
axis you did not look at is the one answer that makes this loop longer, because
it is indistinguishable from a clean axis until round three.

── EACH FINDING CARRIES FOUR THINGS ─────────────────────────────────────────
1. SEVERITY — P1 blocker / P2 should fix / P3 nit.
2. EVERY SITE. If the same mistake appears elsewhere in the repo, list every
   occurrence NOW — grep for it.
3. THE SMALLEST CHANGE THAT RESOLVES IT — concretely enough to apply.
4. WHAT WOULD CHANGE YOUR MIND — the fact that, if true, makes this a non-issue.

── ALREADY SETTLED — do not raise these again in the same form ──────────────
| # | iter | finding | disposition | why |
|---|------|---------|-------------|-----|
| 1 | 1 | `common/collectionRefresh.ts` was in `common/` but nothing in `src/` read it | FIXED | commit cee35741: binding typed against it; the two then disagreed on optional-vs-undefined and the wire shape was corrected to the plugin's. Break-verified: 11 type errors on mutation, previously 0 in the binding |
| 2 | 1 | Should `calendarSyncEnabled` gate the manual Sync button? | REJECTED | the flag is read ONLY by `buildSystemTasks` via config-routes.ts:172; the manual feed refresh does not consult `feedRefreshEnabled` either; the Settings section is titled "Built-in scheduled tasks … Both run hourly"; MulmoClaude has no equivalent setting |
| 3 | 1 | `calendarPush.ts`'s divergence comment cites a `fetchJson` behaviour false since #913 | DEFERRED | prose on a DIFFERENT route; raised as item 6 in the PR body, not changed here |

That table is context so you do not spend this round re-deriving what earlier rounds
already decided. It is NOT a list of closed topics: if a resolution is wrong, or a FIXED
row's fix does not do what it claims, say so and mark it 'REOPENING #<row>' with the
specific evidence. Silence on a bad fix is worse than a repeat finding.

── END WITH ONE TOP-LEVEL COMMENT, IN THIS ORDER ────────────────────────────
  Line 1, a verdict marker on its own line:
    'CODEX VERDICT: LGTM' if you have no outstanding concerns
    'CODEX VERDICT: CHANGES REQUESTED' followed by a bulleted summary of remaining issues
  Then the axis table: one row per axis above, findings or 'none'.
  Then, on its own line:
    'FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.'
  Omit that last line if it is not true, and say which part of the diff you did
  not reach and why.

Do not apply any fixes yourself — only review and post findings.
```

</details>

<details><summary>Codex reply (verbatim stdout)</summary>

```text
Reviewed PR #2111 and posted the required top-level review.

Verdict: LGTM — no findings across any requested axis.

Checks passed:

- Targeted Vitest: 25 tests
- `yarn typecheck`

Review: top-level comment (#issuecomment-5711542112)
```

</details>

Codex's own verdict comment, with the per-axis table and the `FINDINGS COMPLETE` line, is
[#issuecomment-5711542112](https://github.com/receptron/mulmoterminal/pull/2111#issuecomment-5711542112).

**One honest qualification about what it read.** Its stderr shows it picked up
`test/server/backends/feeds.spec.ts | 10 ++++++++++` — an uncommitted test that was in the working
tree while the round ran. So the verdict covers the pushed head plus that test, which has now been
committed as `696ad56a`. Nothing else was uncommitted.

**Comment by isamu** [comment]
| # | iter | finding (one line) | disposition | why / where |
|---|------|--------------------|-------------|-------------|
| 0 | 0 | Sandbox pre-flight: what codex can execute | — | vitest exit 0, `yarn typecheck` exit 0, socket bind exit 0, `curl api.github.com` 200, `gh` works. **File deletion refused by policy** — so no mutation sweep is asked of codex; any mutation it names, I run. |
| 0b | 0 | Pre-existing red baseline | — | CI on this head: knip, jscpd, triage, Socket, CodeRabbit all SUCCESS. CodeRabbit reports "Review rate limited" and has reviewed neither commit — an absence, not a sign-off. Repo's Codex auto-review workflow is deliberately disabled. |
| 1 | 1 | `common/collectionRefresh.ts` claimed both sides decide from it while nothing in `src/` read it | FIXED | commit `cee35741`; binding typed against it, and the resulting disagreement on optional-vs-`undefined` corrected to the plugin's shape. Break-verified: mutating a field gives 11 type errors across server arms + binding, previously 0 in the binding |
| 2 | 1 | Should `calendarSyncEnabled` gate the manual button? | REJECTED | The flag is read ONLY in `server/config/config-routes.ts:172` → `buildSystemTasks`; the manual feed refresh does not consult `feedRefreshEnabled` either; Settings section title is "Built-in scheduled tasks … Both run hourly" (`src/i18n/en.ts`). MulmoClaude has no equivalent setting (grepped `../mulmoclaude`) |
| 3 | 1 | `calendarPush.ts` justifies its 200-on-refusal with a `fetchJson` claim false since `#913` | DEFERRED | Prose on a different route; `e268f5c1` wrote it 10:10, `e20ee14b` (#913) landed 11:03 same day. Raised as item 6 in the PR body |
| 4 | 1 | Codex round 1, tier C: LGTM, all 10 axes answered, FINDINGS COMPLETE present | — | [#issuecomment-5711542112](https://github.com/receptron/mulmoterminal/pull/2111#issuecomment-5711542112). Sandbox verified first (row 0), so "no findings" is a statement about checks it could actually run |
| 5 | 1 | Claude (self-review, not flagged by Codex): the PR body claimed the feed arm's response was unchanged, and that claim was argued rather than run | FIXED | 320-case differential of the old body expression vs the new over generated `written`/`errors`/`dispatched`/`chatId` — identical wire bytes in every case. Generator + property harvested into a permanent test (`696ad56a`); break-verified, the truthiness mutation fails exactly that test |

File: test/server/backends/feeds.spec.ts [hunk 1]
```diff
@@ -83,3 +92,95 @@ describe("DELETE /api/feeds/:slug", () => {
...
+
+  it("syncs a googleCalendar collection and answers the sync's counts", async () => {
+    withSchema({ title: "Cal", googleCalendar: { calendarId: "primary", map: {} } });
+    vi.mocked(syncCalendarCollection).mockResolvedValue({ refreshed: true, written: 3, removed: 1, errors: [] });
+    const res = await refresh();
+    expect(res.status).toBe(200);
+    expect(await res.json()).toEqual({ refreshed: true, written: 3, removed: 1, errors: [] });
+    // The REQUEST's root, not the module's: this host serves several, and the lookup that
+    // admitted the slug resolved the same one.
+    expect(vi.mocked(syncCalendarCollection)).toHaveBeenCalledWith("cal", ws);
```
Comment (isamu) [inline test/server/backends/feeds.spec.ts:126]: P2 — This test does not actually pin the multi-root contract it claims to cover. `initProjectRoots` has no `knownProjects`, the request has no `?project=<id>`, and `ws` is both the default workspace and the expected sync root, so a regression to `loadCollection(req.params.slug)` or `syncCalendarCollection(..., workspacePath)` would still pass. Please add a second temporary project via `knownProjects`, issue the refresh with that project id, and assert both `loadCollection` and `syncCalendarCollection` receive that project root. Every occurrence: this is the only new refresh-route test site that exercises root selection. What would change my mind: an existing integration test covering this endpoint with a non-workspace `?project` root and asserting lookup plus sync use the same root.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED
- P2: Add a real non-workspace project-root case to the refresh-route test. The current test uses the default workspace for both lookup/sync expectations, so it would not go red if either new call regressed to the module workspace; inline comment posted on `test/server/backends/feeds.spec.ts:120`.

| Axis | Result |
|---|---|
| correctness and edge cases | none |
| security (path traversal / SSRF / injection / secrets / project-scope escape) | none; the new path accepts only the existing slug and resolved project scope, and delegates Google/feed safety to the engines |
| tests | finding: the new calendar/ingest outcome tests cover the response behavior, including `dispatched: false`, but do not cover a non-workspace refresh request; see inline P2 |
| API / schema / wire / on-disk compatibility | none; feed JSON bytes remain equivalent, and the shared refresh type matches the plugin-compatible optional-key shape |
| consistency with the rest of the codebase | none |
| accessibility and i18n lockstep | none; no UI copy or accessible markup changed |
| [C] BLAST RADIUS — every collection refresh passes through this route | none; `ingest` retains precedence, `googleCalendar` reaches the new arm, and neither remains a 400 |
| [C] STATUS vs FIELD | none; undeclared is a 400, while calendar setup refusals are 200 with `errors`, matching the sibling refresh implementation and the view’s two error paths |
| [C] MULTI-ROOT CORRECTNESS | implementation none; test coverage finding as above. The production route passes the same request scope to `loadCollection` and the selected refresh arm |
| [C] CROSS-HOST PARITY | none in the changed behavior; route arm, wording, outcome fields, and precedence match MulmoClaude. The documented initial-sync omission is explicitly out of scope, and the merged shapescript changes are unrelated to this PR |

Targeted verification: `npx vitest run test/server/backends/calendarRefresh.spec.ts test/server/backends/calendarRefreshResult.spec.ts test/server/backends/feeds.spec.ts test/server/infra/shapescriptRenderTool.spec.ts` — 29 passed, 3 skipped; `yarn typecheck` — passed.

FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.

**Comment by isamu** [comment]
## Round 2 triage — one P2 accepted, fixed, break-verified

**Codex verdict: CHANGES REQUESTED.** One finding, and it was a good one.

**P2 — the multi-root test did not pin the contract it claimed.** `initProjectRoots` had no
`knownProjects`, the request carried no `?project=`, and `ws` was both the default workspace and
the expected sync root — so reverting either `loadCollection` or `syncCalendarCollection` to a
module-level workspace would have kept every test in the file green.

Codex offered its own falsifier ("an existing integration test covering this endpoint with a
non-workspace `?project` root"), so I checked before accepting: `collectionsProjectScope.spec.ts`
covers `/detail`, `/self-containment` and `/items`, and nothing in `test/` issues a refresh with
`?project=`. The premise holds.

**FIXED** — two tests added, one per call site, issuing the refresh against a registered second
project root and asserting the lookup and the arm both receive it. Break-verified by reverting
each call in turn:

| mutation | result |
| --- | --- |
| sync reverts to a module-level workspace | 1 test fails |
| lookup reverts to a module-level workspace | 2 tests fail |
| restored | all green, tree clean |

Before this change both mutations passed the whole file, which is exactly what Codex said.

**Also in this batch, from my own review** — the claims sweep the loop had made overdue. The plan
file's file list predated two of the loop's own commits, so it named neither
`src/composables/collectionUi.ts` nor `docs/mulmoclaude-parity.md`; and the multi-root decision it
records now cites the specs that hold it, so the claim is checkable rather than asserted.

Round 3 is owed on the resulting head — a fix is reviewed for the first time in the round after it
is pushed.

**Comment by isamu** [comment]
CODEX VERDICT: LGTM

| Axis | Result |
| --- | --- |
| correctness and edge cases | none |
| security (path traversal / SSRF / injection / secrets / project-scope escape) | none |
| tests: does a test go red if the new behaviour breaks | none — targeted Vitest passed (31 passed, 3 skipped); `yarn typecheck` passed |
| API / schema / wire / on-disk compatibility | none — feed wire output remains equivalent, and the shared refresh shape matches the plugin-compatible optional-key contract |
| consistency with the rest of the codebase | none |
| accessibility and i18n lockstep | none |
| [C] BLAST RADIUS — schema shapes not considered | none — `ingest` retains precedence, `googleCalendar` reaches the calendar arm, and collections declaring neither remain rejected |
| [C] STATUS vs FIELD | none — the undeclared case is HTTP 400 like MulmoClaude's refresh route, while calendar setup outcomes are HTTP 200 with `errors` for the button banner, matching the reference behavior |
| [C] MULTI-ROOT CORRECTNESS | none — both collection lookup and the selected refresh arm use the request-resolved root; the new non-workspace tests pin both call sites |
| [C] CROSS-HOST PARITY | none — refresh path, status, wording, precedence, and response fields match MulmoClaude; the unported initial-calendar-sync path is explicitly documented as out of scope |

FINDINGS COMPLETE: I read every hunk of the diff and this is every finding I have.

**Comment by isamu** [comment]
## Codex exchange — round 3 (tier C) — LGTM, loop exits

<details><summary>Codex reply (verbatim stdout)</summary>

```text
Reviewed PR #2111 Round 3 and posted the required top-level verdict: LGTM
(#issuecomment-5711724069).

Checks passed:

- Targeted Vitest: 31 passed, 3 skipped
- `yarn typecheck`
- Full diff and MulmoClaude counterparts reviewed
- No inline findings or outstanding concerns
```

</details>

The per-axis table and the `FINDINGS COMPLETE` line are in
[#issuecomment-5711724069](https://github.com/receptron/mulmoterminal/pull/2111#issuecomment-5711724069).
Its multi-root row now reads "the new non-workspace tests pin both call sites" — which is the round-2
finding answered on its own terms.

**Why this is the exit.** The bar is one clean round on the head that will merge, and round 3
reviewed `16a568f2` with nothing pushed during it. Rounds 1 and 2 do not qualify: round 1 was
followed by a push, and round 2 found a P2.

### What the three rounds cost, and why

| round | verdict | what it found |
| --- | --- | --- |
| 1 | LGTM | nothing. I pushed anyway — a self-found test — so it could not be the exit |
| 2 | CHANGES REQUESTED | one P2, correct: the multi-root tests did not pin what they claimed |
| 3 | LGTM | nothing, on the head that will merge |

Three rounds rather than two because round 1's LGTM was followed by a push of my own. That push was
worth making — it is the `dispatched: false` boundary — but it is the honest reason the loop ran an
extra round, and it is the pattern the skill warns about: an unreviewed fix is unreviewed however
good the reason for it was.

**Reviewer coverage on the final head, stated plainly.** Codex reviewed `16a568f2` and signed off.
**CodeRabbit has reviewed no commit on this PR** — it reported "Review rate limited" on every push,
and the repo's Codex auto-review workflow is deliberately disabled. That absence is not a sign-off.

**Comment by isamu** [comment]
## Final state — green and waiting on a human

CI is fully green on `16a568f2`, which is the exact head Codex signed off on in round 3 (nothing
has been pushed since). `mergeable: MERGEABLE`.

Not merged: that is the author's call, and it has not been given. The PR stays open as it is.

**What a reviewer should know before pressing merge**, in one place:

- **One path was never exercised.** No Google account is linked in the environment used, so the
  `kind: "synced"` branch — real events, the `removed` count, the calendar-group fan-out — is
  covered by specs written against the engine's declared outcome type, not by a real calendar.
  Everything else was driven against a running server. One manual press on a linked account is
  what would close this.
- **Only one reviewer saw the final head.** Codex reviewed it and returned LGTM. CodeRabbit
  reported "Review rate limited" on every push and has read no commit on this PR; this repo's
  Codex auto-review workflow is deliberately disabled. That is an absence of review, not an
  approval, and it is worth a human's eyes accordingly.
- **Item 7 in the PR body is a real finding this PR deliberately does not fix** — the divergence
  comment in `calendarPush.ts` rests on a `fetchJson` behaviour that stopped being true in #913.
  The route works; only the justification is stale.
